### DISCUSSION TITLE
Error: SDES and DTLS-SRTP cannot be enabled at the same time.

### DISCUSSION THREAD

**Discussion by SemiConscious** [fonoster/routr#279] [Q&A]
Hi All - First: what a wonderful set of tools routr is. Thank you.

I am working with routr-one in EC2. I am interested initially in using it for Web<->Phone functionality. I found (in my SIP.js application) that if I call it from Zoiper (ie web->phone) I was getting 

`Wed Jun 19 2024 21:56:36 GMT+0100 (British Summer Time) | sip.Invitation | Failed to set remote offer sdp: SDES and DTLS-SRTP cannot be enabled at the same time.`

Referring to here: https://github.com/sipwise/rtpengine/issues/433 it seems to me that https://github.com/fonoster/routr/blob/main/mods/rtprelay/src/utils.ts#L62, and another case in https://github.com/fonoster/routr/blob/main/mods/rtprelay/src/utils.ts#L77, should be 

`SDES: ["off"]`

 instead of

`SDES: "off"`

I have verified this by creating a docker container based on router-one:latest, with the following line: 

`RUN sed -i 's/SDES: "off"/SDES: ["off"]/g' /service/node_modules/@routr/rtprelay/dist/utils.js`

I verified that it makes the above changes, and it fixes the issue. When I use this docker image instead, the error goes away, as do the 'crypto' entries in the SDP.

I'm happy to create a pull request if that would be useful, but it's not clear to me how to build the router-one image, so can't test. If you can set me straight on that I'm happy to help with a pull request.

All the best
Jim

**Comment by psanders** [comment]
Hey Jim,

I looked at the information you provided, and it makes sense.

Feel free to submit a PR with the proposed changes. 

We will have a chance to build and test everything.

**Comment by psanders** [comment]
I turned the conversation as an issue here → https://github.com/fonoster/routr/issues/280

**Reply by SemiConscious** [reply]
Thank you! I have submitted https://github.com/fonoster/routr/pull/281
